### ISSUE TITLE
persistence: META file grows by 150 bytes per second, per table, rewritten entirely twice everysecond 

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#7971]
### What version of Materialize are you using?

```
materialized v0.9.1-dev (780a805fd)

```

### What was the issue?

The `persist/blob/META` file is being rewritten in its entirety twice every second and it grows by about 150 bytes per second per persisted table. I do not think this rate is sustainable as after 24h the file will be 256Mb , or 512Mb/sec just for the I/O of this single file.

### Is the issue reproducible? If so, please provide reproduction instructions.

Use this script to create 20 tables:

```
for i in range (0,20):
   print(f"CREATE TABLE t{i} (f1 INTEGER);")
   print(f"INSERT INTO t{i} VALUES (1);")
   print()
```

### Please attach any applicable log files.
```
strace -f -F target/release/materialized --experimental --persistent-user-tables --log-file stderr 2>&1  | unbuffer -p grep -F 'write(14' | unbuffer -p grep -v 120 | ts

...
Aug 20 12:34:42 [pid 3547482] write(14, "+\0\0\0\0\0\0\0\24\f\0\0\0\0\0\0\0\36\333\377\362\177\0\0+\0\0\0\0\0\0\0"..., 1461960) = 1461960
Aug 20 12:34:42 [pid 3547482] write(14, "+\0\0\0\0\0\0\0\24\f\0\0\0\0\0\0\0\20\333\377\362\177\0\0+\0\0\0\0\0\0\0"..., 1461960) = 1461960
Aug 20 12:34:42 [pid 3547482] write(14, "+\0\0\0\0\0\0\0\24\f\0\0\0\0\0\0\0\36\333\377\362\177\0\0+\0\0\0\0\0\0\0"..., 1461960 <unfinished ...>
```

**Comment by ruchirK** [comment]
This should be addressed by #7966 (or another approach that does the same thing)

**Comment by danhhz** [comment]
closed by #14250
